### PR TITLE
build: init mimas

### PR DESCRIPTION
**Pull request by mweinelt**
Work in progress configuration for a successor to rhea, because the machine cannot keep up with the compression workload by the large number of builders we employ.

The plan is for an [AX162-R](https://www.hetzner.com/dedicated-rootserver/ax162-r/configurator/#/) at Hetzner.

TODO:
- Check we have all relevant configs and secrets from rhea

### CODE REVIEW

**Comment by mweinelt** [comment]
### nvme-SAMSUNG_MZQL21T9HCJR-00A07_S64GNNFX604905

```
smartctl 7.3 2022-02-28 r5338 [x86_64-linux-6.11.3] (local build)
Copyright (C) 2002-22, Bruce Allen, Christian Franke, www.smartmontools.org

=== START OF INFORMATION SECTION ===
Model Number:                       SAMSUNG MZQL21T9HCJR-00A07
Serial Number:                      S64GNNFX604905
Firmware Version:                   GDC5902Q
PCI Vendor/Subsystem ID:            0x144d
IEEE OUI Identifier:                0x002538
Total NVM Capacity:                 1,920,383,410,176 [1.92 TB]
Unallocated NVM Capacity:           0
Controller ID:                      6
NVMe Version:                       1.4
Number of Namespaces:               32
Namespace 1 Size/Capacity:          1,920,383,410,176 [1.92 TB]
Namespace 1 Utilization:            1,153,474,560 [1.15 GB]
Namespace 1 Formatted LBA Size:     512
Local Time is:                      Tue Dec  3 19:55:52 2024 CET
Firmware Updates (0x17):            3 Slots, Slot 1 R/O, no Reset required
Optional Admin Commands (0x005f):   Security Format Frmw_DL NS_Mngmt Self_Test MI_Snd/Rec
Optional NVM Commands (0x005f):     Comp Wr_Unc DS_Mngmt Wr_Zero Sav/Sel_Feat Timestmp
Log Page Attributes (0x0e):         Cmd_Eff_Lg Ext_Get_Lg Telmtry_Lg
Maximum Data Transfer Size:         512 Pages
Warning  Comp. Temp. Threshold:     80 Celsius
Critical Comp. Temp. Threshold:     83 Celsius
Namespace 1 Features (0x1a):        NA_Fields No_ID_Reuse NP_Fields

Supported Power States
St Op     Max   Active     Idle   RL RT WL WT  Ent_Lat  Ex_Lat
 0 +    25.00W   14.00W       -    0  0  0  0       70      70
 1 +     8.00W    8.00W       -    1  1  1  1       70      70

Supported LBA Sizes (NSID 0x1)
Id Fmt  Data  Metadt  Rel_Perf
 0 +     512       0         0
 1 -    4096       0         0

=== START OF SMART DATA SECTION ===
SMART overall-health self-assessment test result: PASSED

SMART/Health Information (NVMe Log 0x02)
Critical Warning:                   0x00
Temperature:                        38 Celsius
Available Spare:                    100%
Available Spare Threshold:          10%
Percentage Used:                    0%
Data Units Read:                    528,298 [270 GB]
Data Units Written:                 439,541 [225 GB]
Host Read Commands:                 64,232,059
Host Write Commands:                54,331,568
Controller Busy Time:               10
Power Cycles:                       3
Power On Hours:                     6
Unsafe Shutdowns:                   0
Media and Data Integrity Errors:    0
Error Information Log Entries:      2
Warning  Comp. Temperature Time:    0
Critical Comp. Temperature Time:    0
Temperature Sensor 1:               38 Celsius
Temperature Sensor 2:               46 Celsius

Error Information (NVMe Log 0x01, 16 of 64 entries)
No Errors Logged
```

### nvme-SAMSUNG_MZQL21T9HCJR-00A07_S64GNNFX604919

```
smartctl 7.3 2022-02-28 r5338 [x86_64-linux-6.11.3] (local build)
Copyright (C) 2002-22, Bruce Allen, Christian Franke, www.smartmontools.org

=== START OF INFORMATION SECTION ===
Model Number:                       SAMSUNG MZQL21T9HCJR-00A07
Serial Number:                      S64GNNFX604919
Firmware Version:                   GDC5902Q
PCI Vendor/Subsystem ID:            0x144d
IEEE OUI Identifier:                0x002538
Total NVM Capacity:                 1,920,383,410,176 [1.92 TB]
Unallocated NVM Capacity:           0
Controller ID:                      6
NVMe Version:                       1.4
Number of Namespaces:               32
Namespace 1 Size/Capacity:          1,920,383,410,176 [1.92 TB]
Namespace 1 Utilization:            1,153,474,560 [1.15 GB]
Namespace 1 Formatted LBA Size:     512
Local Time is:                      Tue Dec  3 19:55:56 2024 CET
Firmware Updates (0x17):            3 Slots, Slot 1 R/O, no Reset required
Optional Admin Commands (0x005f):   Security Format Frmw_DL NS_Mngmt Self_Test MI_Snd/Rec
Optional NVM Commands (0x005f):     Comp Wr_Unc DS_Mngmt Wr_Zero Sav/Sel_Feat Timestmp
Log Page Attributes (0x0e):         Cmd_Eff_Lg Ext_Get_Lg Telmtry_Lg
Maximum Data Transfer Size:         512 Pages
Warning  Comp. Temp. Threshold:     80 Celsius
Critical Comp. Temp. Threshold:     83 Celsius
Namespace 1 Features (0x1a):        NA_Fields No_ID_Reuse NP_Fields

Supported Power States
St Op     Max   Active     Idle   RL RT WL WT  Ent_Lat  Ex_Lat
 0 +    25.00W   14.00W       -    0  0  0  0       70      70
 1 +     8.00W    8.00W       -    1  1  1  1       70      70

Supported LBA Sizes (NSID 0x1)
Id Fmt  Data  Metadt  Rel_Perf
 0 +     512       0         0
 1 -    4096       0         0

=== START OF SMART DATA SECTION ===
SMART overall-health self-assessment test result: PASSED

SMART/Health Information (NVMe Log 0x02)
Critical Warning:                   0x00
Temperature:                        36 Celsius
Available Spare:                    100%
Available Spare Threshold:          10%
Percentage Used:                    0%
Data Units Read:                    527,834 [270 GB]
Data Units Written:                 440,387 [225 GB]
Host Read Commands:                 64,173,963
Host Write Commands:                54,437,024
Controller Busy Time:               10
Power Cycles:                       3
Power On Hours:                     6
Unsafe Shutdowns:                   0
Media and Data Integrity Errors:    0
Error Information Log Entries:      2
Warning  Comp. Temperature Time:    0
Critical Comp. Temperature Time:    0
Temperature Sensor 1:               36 Celsius
Temperature Sensor 2:               45 Celsius

Error Information (NVMe Log 0x01, 16 of 64 entries)
No Errors Logged
```

**Review by Mic92 — APPROVED** [review]
Passes the smell test.

File: build/mimas/disko.nix [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        type = "zfs_fs";
+        mountpoint = "/";
+      };
+      "nix/store" = {
+        type = "zfs_fs";
+        mountpoint = "/nix";
+      };
+      "nix/db" = {
+        type = "zfs_fs";
+        mountpoint = "/nix/var/nix/db";
```
Comment (Mic92) [inline build/mimas/disko.nix:62]: Do we want different block sizes for postgres?

Comment (mweinelt) [reply build/mimas/disko.nix:62]: This is sqlite.

**Comment by mweinelt** [comment]
Hydra is now running on mimas. I runtime-masked the hydra services on rhea for now. Will get some sleep and check back tomorrow.

File: build/rhea/configuration.nix [hunk 2]
```diff
@@ -1,55 +1,28 @@
...
+    "ssh-ed25519 AAAAC3NzaC1lZDI1NTE5AAAAIIY0EGPGfXD1L+YdSJIKUzeFyuUfVW58kMh+mSflEFx1 root@mimas"
+  ];
+
   system.stateVersion = "21.11";
 
-  services.hydra-dev.dbi = "dbi:Pg:dbname=hydra;host=10.254.1.9;user=hydra;";
   systemd.services.hydra-init = {
     after = [ "wireguard-wg0.service" ];
     requires = [ "wireguard-wg0.service" ];
   };
```
Comment (vcunat) [inline build/rhea/configuration.nix:23]: This `hydra-init` dependency was dropped now?

Comment (mweinelt) [reply build/rhea/configuration.nix:23]: I'm actually not sure what to do with it, given that it didn't work in the first place.

Comment (vcunat) [reply build/rhea/configuration.nix:23]: Oh, OK.  If it didn't work, I'd keep it dropped without replacement in this pull request.  (possibly some TODO comment)

Comment (mweinelt) [reply build/rhea/configuration.nix:23]: We already have #366 filed.

File: build/mimas/default.nix [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+  # garbage collection
+  nix.gc = {
+    automatic = true;
+    options = ''--max-freed "$((400 * 1024**3 - 1024 * $(df -P -k /nix/store | tail -n 1 | ${pkgs.gawk}/bin/awk '{ print $4 }')))"'';
+    dates = "03,09,15,21:15";
+  };
+
+  # gc outputs as well, since they are served from the cache
+  nix.extraOptions = "gc-keep-outputs = false";
```
Comment (SuperSandro2000) [inline build/mimas/default.nix:38]:
```suggestion
  nix.settings.gc-keep-outputs = false;
```

File: build/mimas/disko.nix (hunk 1 above)
Comment (Mic92) [reply build/mimas/disko.nix:62]: Ah. Sorry. I meant the postgres database.

Comment (mweinelt) [reply build/mimas/disko.nix:62]: It is on haumea, another host.

File: build/hydra.nix [hunk 4]
```diff
@@ -93,6 +93,9 @@ in
 
     max_concurrent_evals = 1
 
+    # increase the number of active compress slots (CPU is 48*2 on mimas)
+    max_local_worker_threads = 144
```
Comment (vcunat) [inline build/hydra.nix:97]: Now we had a longer period of "Copying Results" hovering roughly around 130, and PSI kept still very low.  So I believe this certainly still isn't a too high value.  We'll see over time.  So far it's been relatively rare to get so high anyway.

**Pull request merged**